src/journal/sd-journal.c

```diff
@@ -316,7 +317,7 @@ _public_ int sd_journal_add_match(sd_journal *j, const void *data, size_t size)
 
         m->le_hash = le_hash;
         m->size = size;
-        m->data = memdup(data, size);
+        m->data = memdup_suffix0(data, size);
```

Comment:
As far as I know, `strict_string_checks=1` is used mostly to find the code that handles binary data with string functions (which isn't always necessarily buggy). This will silence ASan forever but it will also make it impossible to catch issues like the one fixed in the second commit. I'd rather we didn't add `\0` here.